martin/src/pmtiles/mod.rs

```diff
@@ -146,9 +156,25 @@ impl SourceConfigExtras for PmtConfig {
 
     async fn new_sources_url(&self, id: String, url: Url) -> FileResult<TileInfoSource> {
         match url.scheme() {
-            "s3" => Ok(Box::new(
-                PmtS3Source::new(self.new_cached_source(), id, url).await?,
-            )),
+            "s3" => {
+                let force_path_style = self
+                    .force_path_style
+                    .unwrap_or_else(|| get_env_as_bool("AWS_S3_FORCE_PATH_STYLE"));
+                let require_credentials = self.require_credentials.unwrap_or_else(|| {
+                    get_env_as_bool("AWS_REQUIRE_CREDENTIALS")
+                        || !get_env_as_bool("AWS_NO_CREDENTIALS")
+                });
```

Comment:
unsure about `AWS_NO_CREDENTIALS`.

```suggestion
                let require_credentials = self.require_credentials.unwrap_or_else(|| get_env_as_bool("AWS_REQUIRE_CREDENTIALS"));
```

```suggestion
                // `AWS_NO_CREDENTIALS` was the name in some early documentation of this feature
                let require_credentials = self.require_credentials.unwrap_or_else(|| {
                    get_env_as_bool("AWS_REQUIRE_CREDENTIALS")
                        || !get_env_as_bool("AWS_NO_CREDENTIALS")
                });
```